lib/ansible/modules/windows/win_audit_rule.py

```diff
@@ -0,0 +1,139 @@
+    Log in the Event viewer.
+version_added: "2.5"
+author:
+  - Noah Sparks (@nwsparks)
+options:
+  path:
+    description:
+      - Path to the file, folder, or registry key.
+      - Registry paths should be in Powershell format, beginning with an abbreviation for the root
+        'hklm:\software'.
+    required: true
+    aliases: [ dest, destination ]
+  identity_reference:
+    description:
+      - The user or group to adjust rules for.
+    required: true
+  rights:
+    description:
+      - Comma seperated list of the rights desired. Only required for adding a rule.
+    required: true
+  inheritance_flags:
+    description:
+      - Defines what objects inside of a folder will inherit the settings.
+      - If C(path) is a file or directory, rights can be any right under MSDN
+        FileSystemRights U(https://msdn.microsoft.com/en-us/library/system.security.accesscontrol.filesystemrights=.aspx).
+      - If C(path) is a registry key, rights can be any right under MSDN
+        RegistryRights U(https://msdn.microsoft.com/en-us/library/system.security.accesscontrol.registryrights.aspx).
+    required: true
+    default: "ContainerInherit,ObjectInherit"
+    choices: [ ContainerInherit, ObjectInherit ]
+  propagation_flags:
+    description:
+      - Propagation flag on the audit rules.
+      - For more information on the choices see MSDN PropagationFlags enumeration
+        at U(https://msdn.microsoft.com/en-us/library/system.security.accesscontrol.propagationflags.aspx).
+    required: true
+    default: None
+    choices: [ None, InherityOnly, NoPropagateInherit ]
+  audit_flags:
+    description:
+      - Defines whether to log on failure, success, or both.
+      - To log both define as comma seperated list "Success, Failure"
+    required: true
+    default: Success
+    choices: [ Success, Failure ]
+  state:
+    description:
+      - Whether the rule should be present or absent
+      - For absent, only path, identity_reference, and state are required
+    required: true
+    default: present
+    choices: [ present, absent ]
+'''
+
+EXAMPLES = r'''
+- name: add audit rule
+  win_audit_rule:
+    path: 'c:\inetpub\wwwroot\website'
+    identity_reference: 'BUILTIN\Users'
+    rights: 'write,delete,changepermissions'
+    audit_flags: 'success,failure'
+
+- name: remove audit rule
+  win_audit_rule:
+    path: 'c:\inetpub\wwwroot\website'
+    identity_reference: 'BUILTIN\Users'
+    state: absent
+'''
+
+RETURN = r'''
+current_audit_rules:
+  description:
+    - The current rules on the defined c(path)
+    - Will return null if no rules present
+  returned: always
+  type: dictionary
+  sample: |
+    {
+      "audit_flags": "Success",
+      "identity_reference": "Everyone",
+      "inheritance_flags": "False",
+      "is_inherited": "False",
+      "propagation_flags": "None",
+      "rights": "Delete"
+    }
+identity_reference:
```

Comment:
You will need to remove these and below as we are moving away from setting the invocation parameters.